sklearn/metrics/tests/test_pairwise.py

```diff
@@ -110,6 +111,19 @@ def test_pairwise_distances(global_dtype):
     S2 = haversine_distances(X, Y)
     assert_allclose(S, S2)
 
+    # test paired haversine
+    X = rng.randn(5, 3)
+    Y = rng.randn(5, 3)
+    with pytest.raises(ValueError):
+        paired_haversine_distances(X, Y)
+
+    # compare paired to pairwise implementations
```

Comment:
Isn't it already tested below?
https://github.com/scikit-learn/scikit-learn/blob/348f01ab3f10852906adc168460e4c342a4ae471/sklearn/metrics/tests/test_pairwise.py#L447-L473

Reply:
Nope, the haversine is excluded?

Reply:
If metric is `haversine`, `X` and `Y` are reduced to `X[:,:2]` and `Y[:,:2]` and then tested.
`haversine` is definitely skipped for sparse representation.

Reply:
I'm not sure I get this still. I think even if the same functionality is tested elsewhere, testing different functions still has a purpose?

Reply:
@jona-sassenhagen  you are right: I understand now that thanks to your PR there are now 4 ways to obtain the 'same' function:
1. `haversine_distances(X)`
2. `pairwise_distances(X, metric="haversine")`
3. `paired_distances(X, Y, metric="haversine")`
4. `paired_haversine_distances(X, Y)`

the test `test_pairwise_distances` (from line 62) tests whether 1. is equivalent to 2.  for all the metrics.
the test `test_paired_distances` (from line 440) tests whether 3. is equivalent to 4. for all the metrics.
in the following you are testing if 1. is equivalent to 4.

This is good to have, clearly! Thanks for your patience!

Reply:
Actually `pairwise_distances` and `paired_distances` are different (I let you spot the differences).